### DISCUSSION TITLE
Allow classes and interfaces without a body

### DISCUSSION THREAD

**Discussion by rick-de-water** [dotnet/csharplang#6151] [Language Ideas] [answered]
Records can be defined without a body, while classes and interfaces can not:
```csharp
public record Foo;

public class Bar
{
}

public interface IBar
{
}
```
Why can't we write them without a body similar to records?
```csharp
public record Foo;
public class Bar;
public interface IBar;
```

A practical example from my project:
```csharp
// What I have to write
public interface ITexture : IDeviceObject, IDescribedObject<TextureDescription>, IMemoryBoundObject, IDisposable
{
}

// What I want to write
public interface ITexture : IDeviceObject, IDescribedObject<TextureDescription>, IMemoryBoundObject, IDisposable;
```

**Comment by CyrusNajmabadi** [comment]
We're generally opposed to this due to how rare it is (esp. the class case).  It also may go against another thing users ask for which is allowing `;` for classes, where what follows is still nested inside the class (similar to how namespaces work).  Overall, this is not likely to happen.

**Comment by 333fred** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/5513.
